client/components/TestPlanReportStatusDialog/index.jsx

```diff
@@ -0,0 +1,303 @@
+                <td>{testPlanReport.browser.name}</td>
+                <td>{renderReportStatus(testPlanReport)}</td>
+            </tr>
+        );
+    };
+
+    const getMostRecentTestPlanRun = draftTestPlanRuns => {
+        let allTestResults = draftTestPlanRuns.flatMap(run => run.testResults);
+
+        let mostRecentTestResult = allTestResults.reduce(
+            (latestResult, currentResult) => {
+                let latestDate = new Date(latestResult.completedAt);
+                let currentDate = new Date(currentResult.completedAt);
+                return currentDate > latestDate ? currentResult : latestResult;
+            },
+            { completedAt: new Date(0) }
+        );
+
+        return mostRecentTestResult;
+    };
+
+    const getFormattedDate = dateString => {
+        let date = new Date(dateString);
+        let options = { year: 'numeric', month: 'short', day: 'numeric' };
+        let formattedDate = new Intl.DateTimeFormat('en-US', options).format(
+            date
+        );
+        return formattedDate;
+    };
+
+    const renderCompleteReportStatus = (draftTestPlanRuns, id) => {
+        const mostRecentRun = getMostRecentTestPlanRun(draftTestPlanRuns);
+        const formattedDate = getFormattedDate(mostRecentRun.completedAt);
+        return (
+            <a href={`/report/${testPlanVersion.id}/targets/${id}`}>
+                Report completed on <strong>{formattedDate}</strong>
+            </a>
+        );
+    };
+
+    const renderPartialCompleteReportStatus = (
+        draftTestPlanRuns,
+        percentComplete,
+        metrics
+    ) => {
+        const conflictsCount = metrics.conflictsCount ?? 0;
+        switch (draftTestPlanRuns?.length) {
+            case 0:
+                return <span>In test queue with no testers assigned.</span>;
+            case 1:
+                return (
+                    <span>
+                        {percentComplete}% complete by&nbsp;
+                        <a
+                            href={`https://github.com/${draftTestPlanRuns[0].tester.username}`}
+                        >
+                            {draftTestPlanRuns[0].tester.username}
+                        </a>
+                        &nbsp;with {conflictsCount} conflicts
+                    </span>
+                );
+            default:
+                return (
+                    <span>
+                        {percentComplete}% complete by&nbsp;
+                        {draftTestPlanRuns.length} testers with {conflictsCount}
+                        &nbsp;conflicts
+                    </span>
+                );
+        }
+    };
+
+    const renderReportStatus = testPlanReport => {
+        const { metrics, draftTestPlanRuns, at, browser, id } = testPlanReport;
+        if (metrics) {
+            const percentComplete =
+                calculateTestPlanReportCompletionPercentage(testPlanReport);
+            if (percentComplete === 100) {
```

Comment:
We may want to check for the `testPlanReport.markedFinalAt` property here instead.

And it may also play a little better into the process because of a blocking bug I noticed. I was testing this out by:
1. completing all the results for testers assigned to a row in the Test Queue and then,
2. coming back to this dialog, where I'd now see that the `Report completed on <date>` link was available
3. clicked it and it went to the 404 page

That's because the link won't render unless the test plan report's conflicts are resolved, which the `markedFinalAt` property here would represent.

Reply:
A follow up thought I just had (and verified) is about what happens if a test plan report is marked as final, and the TestPlanVersion is still in the DRAFT phase?

We could theoretically show the report page it generates, with the same "Unapproved Report" disclaimer but I'm unsure if we want to do that, and it doesn't feel like that should work. I can't tell with certainty from the issue description if this is expected, unless I've missed something but If that's the case, the condition must also check that the version's phase is CANDIDATE or RECOMMENDED as well.

Reply:
I check for `markedFinalAt && (phase === RECOMMENDED || phase === CANDIDATE)`. Thanks for the suggestion!